tableauserverclient/server/endpoint/custom_views_endpoint.py

```diff
@@ -0,0 +1,104 @@
+
+    """
+    If the request has no filter parameters: Administrators will see all custom views. 
+        Other users will see only custom views that they own.
+    If the filter parameters include ownerId: Users will see only custom views that they own.
+    If the filter parameters include viewId and/or workbookId, and don't include ownerId:
+        Users will see those custom views that they have Write and WebAuthoring permissions for.
+    If site user visibility is not set to Limited, the Users will see those custom views that are "public",
+     meaning the value of their shared attribute is true.
+    If site user visibility is set to Limited, ????
+    """
+
+    @api(version="3.18")
+    def get(self, req_options: Optional["RequestOptions"] = None) -> Tuple[List[CustomViewItem], PaginationItem]:
+        logger.info("Querying all custom views on site")
+        url = self.baseurl
+        server_response = self.get_request(url, req_options)
+        pagination_item = PaginationItem.from_response(server_response.content, self.parent_srv.namespace)
+        all_view_items = CustomViewItem.list_from_response(server_response.content, self.parent_srv.namespace)
+        return all_view_items, pagination_item
+
+    @api(version="3.18")
+    def get_by_id(self, view_id: str) -> Optional[CustomViewItem]:
+        if not view_id:
+            error = "Custom view item missing ID."
+            raise MissingRequiredFieldError(error)
+        logger.info("Querying custom view (ID: {0})".format(view_id))
+        url = "{0}/{1}".format(self.baseurl, view_id)
+        server_response = self.get_request(url)
+        return CustomViewItem.from_response(server_response.content, self.parent_srv.namespace)
+
+    @api(version="3.18")
+    def populate_image(self, view_item: CustomViewItem, req_options: Optional["ImageRequestOptions"] = None) -> None:
+        if not view_item.id:
+            error = "Custom View item missing ID."
+            raise MissingRequiredFieldError(error)
+
+        def image_fetcher():
+            return self._get_view_image(view_item, req_options)
+
+        view_item._set_image(image_fetcher)
+        logger.info("Populated image for custom view (ID: {0})".format(view_item.id))
+
+    def _get_view_image(self, view_item: CustomViewItem, req_options: Optional["ImageRequestOptions"]) -> bytes:
+        url = "{0}/{1}/image".format(self.baseurl, view_item.id)
+        server_response = self.get_request(url, req_options)
+        image = server_response.content
+        return image
+
+    """
+    Not yet implemented: pdf or csv exports
+    """
+
+    @api(version="3.18")
+    def update(self, view_item: CustomViewItem) -> Optional[CustomViewItem]:
+        if not view_item.id:
+            error = "Custom view item missing ID."
+            raise MissingRequiredFieldError(error)
+        if not (view_item.owner or view_item.name or view_item.shared):
```

Comment:
I don't see anything about updating the `shared` property in the REST API docs. Is the check on `view_item.shared` necessary?

Reply:
It's aspirational. Actually sending a value now wouldn't work, but this means that it will work in tsc as soon as the REST API supports it.

Reply:
Gotcha. But might this condition prevent someone from changing a shared CV to be not-shared?

Reply:
No - this condition is just saying that if the user hasn't set any values on the view_item, don't bother sending an update request. Currently we'll send a request if they change the shared attribute, but that will fail.

Reply:
Hmm, if a user did not specify a value for either `owner` or `name` but set `shared` to `False`, this condition would still be true?